app/src/main/java/pl/gittobefit/database/repository/TrainingRepository.java

```diff
@@ -105,6 +107,17 @@ public Exercise getExercise(long id)
         return trainingsToSend;
     }
 
+    public ArrayList<Training> getTrainingsToSendAfterChanges()
+    {
+        ArrayList<Training> trainingsToSend = new ArrayList<>();
+        ArrayList<TrainingWithForm> trainingsInDB = getAllTrainings();
+        for(TrainingWithForm trainingDB : trainingsInDB)
+        {
+            trainingsToSend.add(new Training(trainingDB, this));
+        }
+        return trainingsToSend;
```

Comment:
Zamieniłbym to na strumienie i mapował ```trainingsInDB``` na ```trainingsToSend```. Nie wiem czy korzystałeś z takowych, więc podrzucam ci przydatne linki:
https://www.geeksforgeeks.org/stream-map-java-examples/ - mapowanie z A na B,
https://www.journaldev.com/32457/java-stream-collect-method-examples - zbieranie rzeczy ze streama do listy